hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -105,6 +107,24 @@ public FileSystemBackedTableMetadata(HoodieEngineContext engineContext, Serializ
     return partitionPaths;
   }
 
+  @Override
+  public Map<String, FileStatus[]> getAllFilesInPartitions(List<String> partitionPaths)
+      throws IOException {
+    if (partitionPaths == null || partitionPaths.isEmpty()) {
+      return Collections.emptyMap();
+    }
+
+    int parallelism = Math.min(DEFAULT_LISTING_PARALLELISM, partitionPaths.size());
+
+    List<Pair<String, FileStatus[]>> partitionToFiles = engineContext.map(partitionPaths, partitionPathStr -> {
+      Path partitionPath = new Path(partitionPathStr);
+      FileSystem fs = partitionPath.getFileSystem(hadoopConf.get());
+      return Pair.of(partitionPathStr, FSUtils.getAllDataFilesInPartition(fs, partitionPath));
```

Comment:
`FileStatus` is not a Serializable class by the default spark serializer. If user have not specify the serializer to kryo when query hudi table, an NotSerializableException will throw out. There is a some problem for `FSUtils.getAllPartitionPaths.` But I think we fix this in another PR.

Reply:
I see yeah FileStatus is not serializable in Hadoop 2, but has been made Serializable in Hadoop 3. We should fix this in a separate PR for all methods by introducing `SerializableFileStatus` similar to Spark https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala#L347.